src/IGit.ts

```diff
@@ -0,0 +1,26 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable, Event, Uri } from "vscode";
+import { APIState, PublishEvent, Repository } from "./git";
+import { Metadata, PostCommitCommandsProvider } from "./rrapi";
+
```

Comment:
The types are actually from here: https://github.com/microsoft/vscode/blob/48eea9e7bc7882a0bc44ac0e4ab6f63df05910f5/extensions/git/src/api/git.d.ts#L284C8-L303

They just call it an API-- it's the same code that used to be here: https://github.com/microsoft/vscode-azurestaticwebapps/blob/main/src/git.d.ts#L255-L272 (which is copied from vscode)

I renamed it to IGit though because RemoteRepos implements the same API and I believe that they implemented it as `IGit`.

That being said, I can still link the vscode `git.d.ts` file.